### PR TITLE
Fix Default Provider for Bitcoin

### PR DESCRIPTION
**Pull request by glethuillier**
Presently, the default provider for Bitcoin is no longer operating as it should.

This update changes the default provider to fix this behavior.

Note: other protocols (Bitcoin Cash for instance) should be addressed by another PR.

### CODE REVIEW

File: src/actions/checkBalance.ts [hunk 1]
```diff
@@ -141,7 +141,7 @@ async function deriveAndScanAddressesByDerivationMode(
         txCounter++;
         display.transientLine(/* delete address as it is not an active one */);
 
-        if (account === 1 || txCounter >= configuration.gap_limit) {
+        if (account === 1 || txCounter >= +configuration.gap_limit) {
```
Comment (meriadec) [inline src/actions/checkBalance.ts]: a bit shady/hacky to use "unary plus" operator there (and that's suprising that Typescript allows comparing Number & String here, but probably that `configuration` typing is broken)

would rather recommend to directly cast as Number in `src/configuration/settings.ts` or to explicitely cast as Number in place here

**Review by meriadec — APPROVED** [review]
Hey @glethuillier :)

kinda blindly approved as there is multiple things mixed in PR (GitHub actions, dependencies update (also a bit concerning that we seem to use **both** a `yarn.lock` **and** an `npm-shrinkwrap.json`), and various typing/logic changes, so it's kinda hard to understand the PR substance (but I trust you :D)

Anyway, safe to merge for us (LES) as AFAIK it's now not anymore used in the stack.

Added a minor comment

File: src/actions/checkBalance.ts (hunk 1 above)
Comment (glethuillier) [reply src/actions/checkBalance.ts]: Good catch! Fixed.

Comment (glethuillier) [reply src/actions/checkBalance.ts]: Thank you @meriadec: I have addresses your concern. You can merge if the changes suit you (I am not authorized to merge by myself).

File: .yarnrc [hunk 2]
```diff
@@ -0,0 +1 @@
+--install.no-lockfile true
```
Comment (meriadec) [inline .yarnrc]: not sure to understand why we should do this
it's bad practice to have two different package manager for the same project, either we use `yarn` (and we drop the `npm-.*` files), either we use `npm` and we drop `yarn.lock`

(in both cases, that doesn't seem related to this PR)

**Review by meriadec — COMMENTED** [review]
would recommend to either entirely remove yarn support or npm support, can't be both

File: .yarnrc (hunk 2 above)
Comment (glethuillier) [reply .yarnrc]: I agree. The README still refers to both of them, so this is confusing. I would recommend to stick to `npm`.

Do you want me to drop yarn support in the context of this PR?

Comment (meriadec) [reply .yarnrc]: Yes, as PR already contains dependency updates etc. I think it's fine :+1: Thanks!

Comment (glethuillier) [reply .yarnrc]: Done.

Successfully tested locally. I am a little bit unsure about the Github Actions (particularly the `key` for `actions/cache@v2`), though.

**Review by meriadec — APPROVED** [review]
